### DISCUSSION TITLE
[Detector Support]: RKNN Orange Pi 5b

### DISCUSSION THREAD

**Discussion by TechsavvyCNC** [blakeblackshear/frigate#10907] [Detector Support]
### Describe the problem you are having

Whenever I try to enable the rknn Detector frigate faults out with errors.

### Version

13.2

### Frigate config file

```yaml
Unable to use rknn Detector  on orange 5b with rk3588s.  Errors out everytime I enable it.  Config works fine without it though.

## Configuration
mqtt:
  enabled: true
  host: 192.168.1.136
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: amcrest
  password: ####
  stats_interval: 60
  
environment_vars:
  PRIVILEGED: true

database:
  path: /media/frigate.db

go2rtc:
  streams:
    backdoor:
    - rtsp://admin:####@192.168.1.147:554/cam/realmonitor?channel=1&subtype=0
    backdoor_sub:
    - rtsp://admin:####@192.168.1.147:554/cam/realmonitor?channel=1&subtype=2

cameras:
  backdoor: # <--- this will be changed to your actual camera later
    enabled: true
    ffmpeg:
      hwaccel_args: preset-rk-h264
      inputs:
      - path: 
          rtsp://admin:#####@192.168.1.147:554/cam/realmonitor?channel=1&subtype=0
        input_args: preset-rtsp-restream
        roles:
        - record
      - path: rtsp:/admin:####@192.168.1.147:554/cam/realmonitor?channel=1&subtype=2
        input_args: preset-rtsp-restream
        roles:
        - detect
    detect:
      width: 1280
      height: 720
      fps: 5
    objects:
      track:
      - person
      - cat
      - truck
      - dog
      
    rtmp:
      enabled: false

motion:
  contour_area: 15
  threshold: 30
  
record:
  enabled: true
  retain:
    days: 1
    # hours:
    # minutes: 
    mode: active_objects
  events:
    retain:
      default: 2
      mode: active_objects 
      objects:
        dog: 2
        person: 2
    pre_capture: 5
    post_capture: 10

detectors:
  rknn:
    type: rknn
    core_mask: 0b011
    
model:
  input_pixel_format: bgr
```

### docker-compose file or Docker CLI command

```yaml
Not sure how to see output of docker container.  I do have portainer addon installed though.
```

### Relevant log output

```shell
2024-04-09 08:54:21.480685090    File "/opt/frigate/frigate/detectors/plugins/rknn.py", line 56, in __init__
2024-04-09 08:54:21.480689465      raise Exception("Make sure to run docker in privileged mode.")
2024-04-09 08:54:21.480692382  Exception: Make sure to run docker in privileged mode.
2024-04-09 08:54:21.480756256  Exception ignored in: <function Rknn.__del__ at 0xffff7da523a0>
2024-04-09 08:54:21.480760339  Traceback (most recent call last):
2024-04-09 08:54:21.480785714    File "/opt/frigate/frigate/detectors/plugins/rknn.py", line 141, in __del__
2024-04-09 08:54:21.481160498      self.rknn.release()
2024-04-09 08:54:21.481197831  AttributeError: 'Rknn' object has no attribute 'rknn'
2024-04-09 08:54:21.508791020  [2024-04-09 03:54:21] frigate.app                    INFO    : Camera processor started for backdoor: 481
2024-04-09 08:54:21.510163869  [2024-04-09 03:54:21] frigate.app                    INFO    : Capture process started for backdoor: 482
2024-04-09 08:54:21.659975050  [2024-04-09 03:54:21] frigate.video                  ERROR   : backdoor: Unable to read frames from ffmpeg process.
2024-04-09 08:54:21.660717619  [2024-04-09 03:54:21] frigate.video                  ERROR   : backdoor: ffmpeg process is not running. exiting capture thread..
```

### Operating system

Other Linux

### Install method

Docker CLI

### Coral version

Other

### Any other information that may be helpful

I'm running armbian and installed Supervised home assistant.  Frigate was installed through hacs then the frigate beta adddon was installed.

**Comment by NickM-27** [comment]
The addon is not supported, the proper configs are not passed to the container.

**Reply by TechsavvyCNC** [reply]
Can the container be edited directly?  I see that even though in my config it has privileged: yes 
It doesn't pass it to to the container.

**Reply by NickM-27** [reply]
running portainer addon it not a supported setup. If you want to use rockchip then addons can not be used at all

**Reply by TechsavvyCNC** [reply]
What dev image still has rknn support?

**Comment by TechsavvyCNC** [comment]
So If I just run it not as a addon it should work?  I just dowload the
latest dev image I think?

***@***.***:680db67674927e1f7be2dfa95b847550e2fa3db87350c1c358325f476a5758c1

**Reply by NickM-27** [reply]
the latest dev image has removed rockchip see https://github.com/blakeblackshear/frigate/pull/10717

**Comment by TechsavvyCNC** [comment]
What was the last image with rockchip still in it?

**Comment by MarcA711** [comment]
Hey @TechsavvyCNC I just stumbled across this. As Nick said, currently it is not possible to use rknn when using the HA addon for HA OS or HA supervised. There will be no addon, until this issue has been solved: https://github.com/home-assistant/supervisor/issues/4863

Currently, the only working way to use HA and frigate with Rockchip HW acceleration is by using setting up and correctly configuring these 3 docker images:
- Home Assistant Container
- Frigate (image `marca711/frigate:latest-rk` from docker hub is the latest stable image)
- Mosquitto

**Comment by TechsavvyCNC** [comment]
Don't I have the home assistant and mosquitto containers already
installed?

**Reply by MarcA711** [reply]
In your initial post you said that you use HA supervised, which is something different than HA container (see https://www.home-assistant.io/installation/#advanced-installation-methods)

HA container does not support add-ons. However, most add-ons are also available as standalone docker images, so this is usually no problem. But you need to set them up and configure them properly, which is a bit more difficult than just installing a HA Add-on. If you need more assistance, I am happy to help.
